lib/ical_filter_proxy/calendar_builder.rb

```diff
@@ -33,5 +34,36 @@ def add_rules
       end
     end
 
+    def add_alarms
+      alarms = calendar_config["alarms"]
+      return unless alarms
+
+      if alarms["clear_existing"] == true
+        calendar.set_clear_existing_alarms
+      end
+
+      triggers = alarms["triggers"]
+      return unless triggers
+
+      triggers.each do |trigger|
+        calendar.add_alarm_trigger(generate_iso_format(trigger))
+        end
+    end
+
+    def generate_iso_format(input)
+      case input
+      when /(\d+)\s+days?/i
+        return "-P#{$1}D"
```

Comment:
You don't need the explicit returns, ruby will bail out of the case statement here and it'll be returned implicitly

Reply:
allright, adjusted